config/config_test.go

```diff
@@ -0,0 +1,30 @@
+package config_test
+
+import (
+	"testing"
+
+	"github.com/pganalyze/collector/config"
+)
+
+type testItem struct {
+	input    string
+	expected string
+}
+
+var tests = []testItem{
+	{"postgres://user:password@example.com", "postgres://user@example.com"},
+	{"postgres://user@example.com", "postgres://user@example.com"},
+	{string([]byte{0x7f}), "<unparsable>"},
```

Comment:
@lfittl so this is insanely contrived, but I literally could not find another string to parse that returned an error from `net/url.Parse`. It seems to be designed for also parsing relative URL and so just URL-encodes everything it doesn't recognize: https://golang.org/pkg/net/url/#Parse

Thoughts?

Reply:
Hah. Yeah, I think this works - really the most important cases are the first two.

I wonder if we should have a test case for something that is like `postgres://user:pass:word@example.com`?

Also probably good to have a longer example URL that includes things like `?sslmode=verify-full` and similar.